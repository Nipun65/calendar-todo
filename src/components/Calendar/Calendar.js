import { Fragment, useEffect, useState } from "react";
import Daydate from "../Daydate/Daydate";
import Header from "../Header/Header";
import Card from "../Card/Card";
import styles from "./Calendar.module.css";
import {
  handleMonth,
  handleOption,
  handleToday,
  handleValue,
  handleView,
} from "../Handlers/Handlers";

const Calendar = () => {
  const todayDate = new Date();
  const [date, setDate] = useState({
    year: todayDate.getFullYear(),
    month: todayDate.getMonth(),
    date: todayDate.getDate(),
    currentYear: todayDate.getFullYear(),
    currentMonth: todayDate.getMonth(),
    selectedDate: {
      userYear: 0,
      userMonth: 0,
      userDate: 0,
    },
  });

  const Month = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
  ];

  const [years, setYears] = useState();
  useEffect((value) => {
    const yearsArray = [];
    let Currentyear = date.year;
    let CurrentyearCopy = date.year - (date.year % 10) + 1;
    while (CurrentyearCopy < Currentyear) {
      yearsArray.push(CurrentyearCopy);
      CurrentyearCopy++;
    }
    while (Currentyear % 10 !== 0) {
      yearsArray.push(Currentyear);
      Currentyear++;
      if (Currentyear % 10 === 0) {
        yearsArray.push(Currentyear);
        setYears(yearsArray);
      }
    }
  }, []);

  const [selectedOption, setSelectedOption] = useState("Month");

  const renderElements = (value, index) => {
    const dateObj = {
      year: date.year,
      month: index,
      date: date.date,
      currentMonth: todayDate.getMonth(),
      currentYear: todayDate.getFullYear(),
      selectedDate: {
        userYear: date.selectedDate.userYear,
        userMonth: date.selectedDate.userMonth,
        userDate: date.selectedDate.userDate,
      },
    };
    return (
      <Daydate
        className={styles["grid-item"]}
        setValue={(value) =>
          handleValue(date, view, years, setDate, setYears, value)
        }
        date={dateObj}
        month={Month}
        selectedOption={selectedOption}
        key={Math.random()}
      />
    );
  };

  const [view, setView] = useState("date");

  return (
    <Fragment>
      <Header
        setValue={(value) =>
          handleValue(date, view, years, setDate, setYears, value)
        }
        setView={(value) => handleView(value, setView, date, setYears, years)}
        date={date}
        view={view}
        selectedOption={selectedOption}
        setSelectedOption={(value) =>
          handleOption(setSelectedOption, setView, setDate, date, value)
        }
        setToday={() => handleToday(date, todayDate, setView, setDate)}
      />

      {view === "date" && selectedOption !== "Year" && (
        <Daydate
          setValue={(value) =>
            handleValue(date, view, years, setDate, setYears, value)
          }
          date={date}
          selectedOption={selectedOption}
        />
      )}

      {selectedOption === "Year" && (
        <div className={styles["grid-container-year"]}>
          {Array(12)
            .fill(0)
            .map((value, index) => renderElements(value, index))}
        </div>
      )}
      {view !== "date" && (
        <Card
          data={view === "months" ? Month : years}
          setValue={(value) =>
            handleMonth(
              value,
              Month,
              date,
              setDate,
              setView,
              view,
              years,
              setYears
            )
          }
        />
      )}
    </Fragment>
  );
};

export default Calendar;
