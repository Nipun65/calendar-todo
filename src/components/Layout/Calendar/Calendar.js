import { useEffect, useState } from 'react';
import { useNavigate, useLocation } from 'react-router-dom';
import Daydate from '../../Common/Daydate/Daydate';
import Header from '../../Common/Header/Header';
import Card from '../../Common/Card/Card';
import styles from './Calendar.module.css';
import Todo from '../../Common/Todo/Todo';
import { MONTHS } from '../../../utils/Constants.utils';

function Calendar() {
  const todayDate = new Date();
  const [date, setDate] = useState({
    year: todayDate.getFullYear(),
    month: todayDate.getMonth(),
    date: todayDate.getDate(),
    currentYear: todayDate.getFullYear(),
    currentMonth: todayDate.getMonth(),
    selectedDate: {
      userYear: 0,
      userMonth: 0,
      userDate: 0,
    },
  });
  const [years, setYears] = useState();
  const [selectedOption, setSelectedOption] = useState('Month');
  const [view, setView] = useState('date');
  const history = useNavigate();
  const location = useLocation();

  // updates the view and years state. if view is multiyears then years state contains all decade years. in years view it contains one decade
  const handleView = (value) => {
    setView(value);
    let currentYear = 0;
    if (value === 'multiyears') {
      let startYear = 0;
      if (date.year % 100 !== 0) {
        startYear = date.year - (date.year % 100) + 1;
      } else {
        startYear = date.year - 100 + 1;
      }
      const yearsArray = [];
      let string = '';
      for (let i = 0; i < 10; i++) {
        string = `${startYear} - ${startYear + 9}`;
        yearsArray.push(string);
        startYear += 10;
      }
      setYears([...yearsArray]);
    } else if (value === 'years') {
      if (date.year % 10 !== 0) {
        currentYear = date.year - (date.year % 10) + 1;
      } else {
        currentYear = date.year - 9;
      }

      for (let i = 0; i < 10; i++) {
        years[i] = currentYear;
        currentYear++;
      }
    }
  };

  // validate user entered url and set date according to it
  const validateUrl = (locationObj) => {
    const pathnamePart = locationObj.pathname.split('/');
    if (pathnamePart.length === 2) {
      if (locationObj.search === '') {
        if (pathnamePart[1] === 'month' || pathnamePart[1] === 'year') {
          let newView = '';
          if (pathnamePart[1] === 'month') {
            newView = 'date';
          } else if (pathnamePart[1] === 'year') {
            newView = 'months';
          }
          setView(newView);
        } else if (pathnamePart[1] === 'todo') {
          setView('day');
          date.selectedDate.userDate = todayDate.getDate();
          date.selectedDate.userMonth = todayDate.getMonth();
          date.selectedDate.userYear = todayDate.getFullYear();
        }
      } else if (locationObj.search.length) {
        const secondpart = locationObj.search.split('?');
        const secondpartdivide = secondpart[1].split('=');
        const splitMonthYear = secondpartdivide[1].split('+');
        if (secondpartdivide[0] === pathnamePart[1]) {
          if (pathnamePart[1] === 'month') {
            if (
              splitMonthYear.length === 2 &&
              +splitMonthYear[0] > 0 &&
              +splitMonthYear[0] <= 12 &&
              +splitMonthYear[1] > 0
            ) {
              date.month = +splitMonthYear[0] - 1;
              date.year = +splitMonthYear[1];
            }
          } else if (pathnamePart[1] === 'year') {
            if (splitMonthYear.length === 1 && +splitMonthYear[0] > 0) {
              date.year = +splitMonthYear[0];
              if (selectedOption === 'Month') {
                setView('months');
              }
            }
          } else if (pathnamePart[1] === 'multiyear') {
            const splitsplitMonthYear = splitMonthYear[0].split('-');
            const startYear = +splitsplitMonthYear[0].split('(')[1];
            const endYear = +splitsplitMonthYear[1].split(')')[0];
            if (endYear - startYear === 9 && startYear % 10 === 1) {
              date.year = startYear;
              setView('years');
            } else if (endYear - startYear === 99 && startYear % 100 === 1) {
              date.year = startYear;
              handleView('multiyears');
            }
          }
        } else if (
          pathnamePart[1] === 'todo' &&
          secondpartdivide[0] === 'day'
        ) {
          if (
            splitMonthYear.length === 3 &&
            +splitMonthYear[1] > 0 &&
            +splitMonthYear[1] <= 12
          ) {
            const endDayOfTheMonth = new Date(
              +splitMonthYear[2],
              +splitMonthYear[1],
              0
            );
            if (
              +splitMonthYear[0] >= 1 &&
              splitMonthYear[0] <= endDayOfTheMonth.getDate()
            ) {
              date.selectedDate.userDate = +splitMonthYear[0];
              date.month = +splitMonthYear[1] - 1;
              date.year = +splitMonthYear[2];
              date.selectedDate.userMonth = +splitMonthYear[1] - 1;
              date.selectedDate.userYear = +splitMonthYear[2];
              setView('day');
            }
          }
        }
      }
    }
  };

  // set the years array when first time component render it will be used in Card component years view to show years and validate browser url if the url is not correct it'll redirect to today's date
  useEffect(() => {
    const yearsArray = [];
    let Currentyear = date.year;
    let CurrentyearCopy = date.year - (date.year % 10) + 1;
    while (CurrentyearCopy < Currentyear) {
      yearsArray.push(CurrentyearCopy);
      CurrentyearCopy++;
    }
    while (Currentyear % 10 !== 0) {
      yearsArray.push(Currentyear);
      Currentyear++;
      if (Currentyear % 10 === 0) {
        yearsArray.push(Currentyear);
        setYears(yearsArray);
      }
    }
    validateUrl(location);
  }, []);

  // handle navigation as per view
  useEffect(() => {
    let navigateTo = '';
    if (view === 'date') {
      if (
        date.month === todayDate.getMonth() &&
        date.year === todayDate.getFullYear()
      ) {
        if (selectedOption === 'Month') {
          navigateTo = '/month';
        } else {
          navigateTo = '/year';
        }
      } else if (selectedOption === 'Year') {
        navigateTo = `/year?year=${date.year}`;
      } else {
        navigateTo = `/month?month=${date.month + 1}+${date.year}`;
      }
    }
    if (view === 'months') {
      if (date.year === todayDate.getFullYear()) {
        navigateTo = '/year';
      } else {
        navigateTo = `/year?year=${date.year}`;
      }
    }
    if (view === 'years') {
      navigateTo = `/multiyear?multiyear=(${date.year - (date.year % 10) + 1}-${
        date.year - (date.year % 10) + 10
      })`;
    }
    if (view === 'multiyears') {
      navigateTo = `/multiyear?multiyear=(${
        date.year - (date.year % 100) + 1
      }-${date.year - (date.year % 100) + 100})`;
    }
    if (view === 'day') {
      if (
        date.selectedDate.userDate === todayDate.getDate() &&
        date.selectedDate.userMonth === todayDate.getMonth() &&
        date.selectedDate.userYear === todayDate.getFullYear()
      ) {
        navigateTo = '/todo';
      } else {
        navigateTo = `/todo?day=${date.selectedDate.userDate}+${
          date.month + 1
        }+${date.year}`;
      }
    }
    history(navigateTo);
  }, [view, date.month, date.year, date.selectedDate.userDate, selectedOption]);

  // updating the years array according to clicked event and view
  const handleValue = (value) => {
    date.year = +date.year;
    if (view === 'multiyears') {
      let startYear = '';
      if (value === 'singleforward') {
        startYear = date.year - (date.year % 100) + 100;
      } else if (value === 'singlebackward') {
        startYear = date.year - (date.year % 100) - 100;
      }

      let string = '';
      for (let i = 0; i < 10; i++) {
        string = `${startYear + 1} - ${startYear + 10}`;
        years[i] = string;
        startYear += 10;
      }

      date.year = +years[0].split('-')[0];
      setDate({ ...date });
      setYears([...years]);
    } else if (view === 'years') {
      for (let i = 0; i < years.length; i++) {
        if (value === 'doubleforward') {
          years[i] += 100;
        } else if (value === 'doublebackward') {
          years[i] -= 100;
        } else if (value === 'singleforward') {
          years[i] += 10;
        } else if (value === 'singlebackward') {
          years[i] -= 10;
        }
      }
      const [firstYear] = years;
      date.year = firstYear;
      setDate({ ...date });
    } else if (view === 'months') {
      if (value === 'doubleforward') {
        date.year += 10;
      } else if (value === 'doublebackward') {
        date.year -= 10;
      } else if (value === 'singleforward') {
        date.year += 1;
      } else if (value === 'singlebackward') {
        date.year -= 1;
      }
      setDate({ ...date });
    } else if (view === 'day') {
      if (value === 'doubleforward') {
        if (date.selectedDate.userMonth === 11) {
          date.selectedDate.userMonth = 0;
          date.selectedDate.userYear += 1;
        } else {
          date.selectedDate.userMonth += 1;
        }
      } else if (value === 'doublebackward') {
        if (date.selectedDate.userMonth === 0) {
          date.selectedDate.userMonth = 11;
          date.selectedDate.userYear -= 1;
        } else {
          date.selectedDate.userMonth -= 1;
        }
      } else if (value === 'singleforward') {
        const lastDay = new Date(
          date.selectedDate.userYear,
          date.selectedDate.userMonth + 1,
          0
        );
        if (date.selectedDate.userDate === lastDay.getDate()) {
          if (date.selectedDate.userMonth === 11) {
            date.selectedDate.userDate = 1;
            date.selectedDate.userMonth = 0;
            date.selectedDate.userYear += 1;
          } else {
            date.selectedDate.userDate = 1;
            date.selectedDate.userMonth += 1;
          }
        } else {
          date.selectedDate.userDate += 1;
        }
      } else if (value === 'singlebackward') {
        if (date.selectedDate.userDate === 1) {
          if (date.month === 0) {
            const lastDay = new Date(
              date.selectedDate.userYear,
              date.selectedDate.userMonth + 1,
              0
            );
            date.selectedDate.userYear -= 1;
            date.selectedDate.userMonth = 11;
            date.selectedDate.userDate = lastDay.getDate();
          } else {
            const lastDay = new Date(
              date.selectedDate.userYear,
              date.selectedDate.userMonth,
              0
            );
            date.selectedDate.userDate = lastDay.getDate();
            date.selectedDate.userMonth -= 1;
          }
        } else {
          date.selectedDate.userDate -= 1;
        }
      }
      date.month = date.selectedDate.userMonth;
      date.year = date.selectedDate.userYear;
      setDate({ ...date });
    } else {
      setDate({ ...value });
    }
  };

  // handle selected month or year in card
  const handleMonthorYear = (value) => {
    const monthIndex = MONTHS.findIndex((ele) => ele === value);
    if (monthIndex !== -1) {
      date.month = monthIndex;
      setDate({ ...date });
      setView('date');
    } else if (view === 'years') {
      date.year = value;
      setDate({ ...date });
      if (selectedOption !== 'Year') {
        setView('months');
      } else {
        setView('date');
      }
    } else if (view === 'multiyears') {
      let startYear = +value.split('-')[0];
      date.year = startYear;
      setDate({ ...date });
      for (let i = 0; i < 10; i++) {
        years[i] = startYear;
        startYear++;
      }
      setView('years');
      setYears(years);
    }
  };

  // to handle today button
  const handleToday = () => {
    date.year = todayDate.getFullYear();
    date.month = todayDate.getMonth();
    date.date = todayDate.getDate();
    date.selectedDate.userYear = todayDate.getFullYear();
    date.selectedDate.userMonth = todayDate.getMonth();
    date.selectedDate.userDate = todayDate.getDate();

    if (view !== 'day') {
      setView('date');
    }
    setDate({ ...date });
  };

  // to handle selected option in dropdown
  const handleOption = (value) => {
    setSelectedOption(value);
    setView('date');
    setDate({ ...date });
  };

  // function for rendering elements in case of selected option is year in dropdown
  const renderElements = (value, index) => {
    const dateObj = {
      year: date.year,
      month: index,
      date: date.date,
      currentMonth: todayDate.getMonth(),
      currentYear: todayDate.getFullYear(),
      selectedDate: {
        userYear: date.selectedDate.userYear,
        userMonth: date.selectedDate.userMonth,
        userDate: date.selectedDate.userDate,
      },
    };
    return (
      <Daydate
        className={styles['grid-item']}
        setValue={handleValue}
        date={dateObj}
        month={MONTHS}
        selectedOption={selectedOption}
        key={Math.random()}
        setView={handleView}
      />
    );
  };

  return (
    <>
      <Header
        setValue={handleValue}
        setView={handleView}
        setSelectedOption={handleOption}
        setToday={handleToday}
        date={date}
        view={view}
        selectedOption={selectedOption}
      />

      {view === 'date' && selectedOption !== 'Year' && (
        <Daydate
          setValue={handleValue}
          setView={handleView}
          date={date}
          selectedOption={selectedOption}
        />
      )}

      {selectedOption === 'Year' && view === 'date' && (
        <div className={styles['grid-container-year']}>
          {Array(12)
            .fill(0)
            .map((value, index) => renderElements(value, index))}
        </div>
      )}
      {/* Card component renders the data. if it's months view then it's will show all months and years view it will show years state */}
      {view !== 'date' && view !== 'day' && (
        <Card
          data={view === 'months' ? MONTHS : years}
          setValue={handleMonthorYear}
        />
      )}
      {view === 'day' && <Todo date={date} Month={MONTHS} />}
    </>
  );
}

export default Calendar;
