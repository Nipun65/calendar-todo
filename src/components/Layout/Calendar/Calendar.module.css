.grid-container-year {
  display: grid;
  grid-template-rows: repeat(4, 1fr);
  grid-template-columns: repeat(3, 1fr);
}

.grid-item {
  border: 1px solid black;
  text-align: center;
}

@media only screen and (max-width: 1100px) {
  .grid-container-year {
    display: grid;
    grid-template-rows: repeat(4, 1fr);
    grid-template-columns: repeat(2, 1fr);
  }
}
@media (max-width: 800px) {
  .grid-container-year {
    display: grid;
    grid-template-rows: repeat(4, 1fr);
    grid-template-columns: repeat(1, 1fr);
  }
}
