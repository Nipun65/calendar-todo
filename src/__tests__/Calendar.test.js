import renderer from 'react-test-renderer';
import Calendar from '../components/Layout/Calendar/Calendar';

it('should match the snapshot', () => {
  const tree = renderer.create(<Calendar />).toJSON();
  expect(tree).toMatchSnapshot();
});
