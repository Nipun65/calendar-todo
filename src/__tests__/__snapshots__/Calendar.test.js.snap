// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
Array [
  <div
    className="header"
  >
    <div
      className="btnHover first-div"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        &lt;&lt;
      </button>
      <button
        className="btn"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        &lt;
      </button>
      <div
        className="todaybtn-div"
      >
        <button
          className="todaybtn"
          onClick={[Function]}
          type="button"
        >
          Today
        </button>
      </div>
    </div>
    <div
      className="
         btnHover second-div"
    >
      <button
        className="btn"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        January 2023 
      </button>
    </div>
    <div
      className="third-div btnHover"
    >
      <div
        className="dropdown"
      >
        <div
          className="dropdown"
        >
          <button
            className="titlebutton"
            onClick={[Function]}
            type="button"
          >
            Month
          </button>
        </div>
      </div>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        &gt;
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        &gt;&gt;
      </button>
    </div>
  </div>,
  <div
    className="main-div "
  >
    <div
      className="days"
    >
      <div
        className="days-ele"
      >
        Sun
      </div>
      <div
        className="days-ele"
      >
        Mon
      </div>
      <div
        className="days-ele"
      >
        Tue
      </div>
      <div
        className="days-ele"
      >
        Wed
      </div>
      <div
        className="days-ele"
      >
        Thu
      </div>
      <div
        className="days-ele"
      >
        Fri
      </div>
      <div
        className="days-ele"
      >
        Sat
      </div>
    </div>
    <div>
      <div
        className="array"
      >
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          1
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          2
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          3
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          4
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          5
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          6
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          7
        </div>
      </div>
      <div
        className="array"
      >
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          8
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          9
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          10
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          11
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          12
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          13
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          14
        </div>
      </div>
      <div
        className="array"
      >
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          15
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          16
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          17
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={
            Object {
              "backgroundColor": "#81d5ea",
            }
          }
        >
          18
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          19
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          20
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          21
        </div>
      </div>
      <div
        className="array"
      >
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          22
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          23
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          24
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          25
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          26
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          27
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          28
        </div>
      </div>
      <div
        className="array"
      >
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          29
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          30
        </div>
        <div
          className="array-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          31
        </div>
        <div
          className="array-element"
        />
        <div
          className="array-element"
        />
        <div
          className="array-element"
        />
        <div
          className="array-element"
        />
      </div>
      <div
        className="array"
      >
        <div
          className="array-element"
        />
        <div
          className="array-element"
        />
        <div
          className="array-element"
        />
        <div
          className="array-element"
        />
        <div
          className="array-element"
        />
        <div
          className="array-element"
        />
        <div
          className="array-element"
        />
      </div>
    </div>
  </div>,
]
`;
